test/test_utils.sh

```diff
@@ -190,29 +192,64 @@ helm_install_fission() {
 wait_for_service() {
     id=$1
     svc=$2
+    health_endpoint=$3
 
     ns=f-$id
+    retry=0
+    max_retries=5
     while true
     do
-	ip=$(kubectl -n $ns get svc $svc -o jsonpath='{...ip}')
-	if [ ! -z $ip ]
-	then
-	    break
-	fi
-	echo Waiting for service $svc...
-	sleep 1
+        retry=$((retry+1))
+        if ((retry == max_retries)); then
+            echo "Waiting for $svc to be routable exceeded max retries. Quitting.."
+            exit 1
+        fi
+        ip=$(kubectl -n $ns get svc $svc -o jsonpath='{...ip}')
+        if [ -z $ip ]; then
+            continue
+        fi
+        http_status=`curl -sw "%{http_code}" "http://$ip/$health_endpoint"`
```

Comment:
👍 this is a good thing to do but I'm curious if it ever fails. (Since helm --wait should be waiting for pods to get ready, and the pods are only ready after http is up because of the readiness probes.)

Reply:
yes that's true, but readiness probe relies on `PodIP` and this function ensures the service `externalIPs` are indeed routing to the pods correctly.